src/main/java/com/mpatric/mp3agic/ID3v2TXXXFrameData.java

```diff
@@ -0,0 +1,228 @@
+package com.mpatric.mp3agic;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ID3v2TXXXFrameData extends AbstractID3v2FrameData {
+
+	protected static final String TAG = "ID3::TXXX";
+
+	public static final String ID_FIELD = "TXXX";
+
+	private EncodedText description;
+	private EncodedText value;
+
+	public ID3v2TXXXFrameData(boolean unsynchronisation) {
+		super(unsynchronisation);
+	}
+
+	public ID3v2TXXXFrameData(boolean unsynchronisation, EncodedText description, EncodedText value) {
+		super(unsynchronisation);
+		this.description = description;
+		this.value = value;
+	}
+
+	public ID3v2TXXXFrameData(boolean unsynchronisation, byte[] bytes) throws InvalidDataException {
+		super(unsynchronisation);
+		synchroniseAndUnpackFrameData(bytes);
+	}
+
+	@Override
+	protected void unpackFrameData(byte[] bytes) throws InvalidDataException {
+		int marker = BufferTools.indexOfTerminatorForEncoding(bytes, 1, bytes[0]);
+
+		description = new EncodedText(bytes[0], BufferTools.copyBuffer(bytes, 1, marker - 1));
```

Comment:
If the description has zero length, the marker position could be 0, in which case the `copyBuffer` will fail. I suggest adding something like what is done in `ID3v2UrlFrameData`, which checks if marker >= 0 before doing the copy, otherwise sets the description to an empty string.